### PR TITLE
Extend ExtendableContext instead of BaseContext for PolicyContext

### LINKED ISSUE #16423: PolicyContext type doesn't include request
### Required System information

- Node.js version: v18.16.0
- NPM version: 9.5.1
- Strapi version: 4.9.1
- Database: MySQL
- Operating system: Mac
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When creating a new custom policy, the type for `PolicyContext` doesn't include keys that should be available like `request`
```typescript
import type { Policy } from "@strapi/strapi/lib/core/registries/policies";

const policy: Policy = (policyContext, config, { strapi }) => {
    //  error TS2339: Property 'request' does not exist on type 'PolicyContext'.
    console.log(policyContext.request);
    
    // ...

    return false;
};

export default policy;
```

### Steps to reproduce the behavior

1. Create a new file at: `./src/api/[apiName]/policies/is-authorized.ts`
2. Copy the above code
3. (optional) Add policy into the routes file at `./src/api/[apiName]/routes/[apiName].ts`

### Expected behavior

There are no Typescript errors and I can access the request object

### Screenshots

N/A

### Code snippets
Example with policyContext keys logged
```typescript
import type { Policy } from "@strapi/strapi/lib/core/registries/policies";

const policy: Policy = (policyContext, config, { strapi }) => {
  // logs this: ["is","type","request","response","app","req","res","originalUrl","state","matched","router","_matchedRoute","captures","params","routerPath","routerName"];
  console.log(Object.keys(policyContext));

  //  error TS2339: Property 'request' does not exist on type 'PolicyContext'.
  console.log(policyContext.request);
  return false;
};

export default policy;
```

### Additional context

[Koa's](https://koajs.com/#context) documentation references request/response. Could be that [PolicyContext](https://github.com/strapi/strapi/blob/294b0f6cfaabd423569ed1d19c6a428a41f45336/packages/core/strapi/lib/core/registries/policies.d.ts#L4) should extend Koa's [ExtendableContext](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/koa/index.d.ts#L720) instead?

Happy to create the PR if that's the case 😄

### PR DESCRIPTION
**Pull request by bmpickford**
### What does it do?

Use [ExtendableContext](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/koa/index.d.ts#L720) instead of BaseContext

### Why is it needed?

Aligns the type with the functionality

### How to test it?

Confirm the type is fixed by using `PolicyContext` and you can use `request` and `response` without Typescript throwing an error

### Related issue(s)/PR(s)

Fix #16423

### CODE REVIEW

File: packages/core/strapi/lib/core/registries/policies.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { BaseContext } from 'koa';
+import { ExtendableContext } from 'koa';
 import { Strapi } from '../../';
 
-interface PolicyContext extends BaseContext {
+interface PolicyContext extends ExtendableContext {
   type: string;
   is(name): boolean;
```
Comment (Convly) [inline packages/core/strapi/lib/core/registries/policies.d.ts:6]:
```suggestion
  is(name: string): boolean;
```

**Review by Convly — COMMENTED** [review]
Thanks for opening this PR!

I just added a comment related to a possible second improvement we could make to the interface.
The rest looks good to me

Note for other reviewers: Since `koa.ExtendableContext` extends `koa.BaseContext`, this won't be a breaking change.

File: packages/core/strapi/lib/core/registries/policies.d.ts (hunk 1 above)
Comment (bmpickford) [reply packages/core/strapi/lib/core/registries/policies.d.ts]: done!

**Review by Convly — APPROVED** [review]
LGTM, waiting for the tests to pass before merging.

Thanks for your contributions!

**Pull request merged**